### PR TITLE
[RESTEASY-2406] Incorporate MicroProfile Config into RESTEasy

### PR DESCRIPTION
**Pull request by ronsigal**

### CODE REVIEW

**Comment by ronsigal** [comment]
Still needs update to RESTEasy User Guide.

File: docbook/reference/en/en-US/modules/Installation_Configuration.xml [hunk 1]
```diff
@@ -442,9 +442,299 @@ public class MyApplication extends Application
...
+        <para>
+            These can be registered programmatically by using an instance of <classname>ConfigProviderResolver</classname>:
+        </para>
+        <programlisting>
+Config config = new PropertiesConfigSource("file:/// ...");
+ConfigProviderResolver.instance().registerConfig(config, getClass().getClassLoader());
+        </programlisting>
+        <para>where <classname>ConfigProviderResolver</classname> is part of the Eclipse MicroProfile Config API.</para>
+        <para>
+            If the application is running in Wildfly, then Wildfly provides another set of <classname>ConfigSource</classname>s,
```
Comment (asoldano) [inline docbook/reference/en/en-US/modules/Installation_Configuration.xml:513]: s/Wildfly/WildFly/

File: docbook/reference/en/en-US/modules/Installation_Configuration.xml [hunk 2]
```diff
@@ -442,9 +442,299 @@ public class MyApplication extends Application
...
+                return "system" rather than "context".
+            </para>
+
+        </section>
+        
+        <section id="configuring_mp_config">
+            <title>Configuring MicroProfile Config</title>
+            
+            <para>
+                If an application is running inside Wildfly, then the MicroProfile Config dependencies are available
```
Comment (asoldano) [inline docbook/reference/en/en-US/modules/Installation_Configuration.xml:591]: s/Wildfly/WildFly/

File: docbook/reference/en/en-US/modules/Installation_Configuration.xml [hunk 3]
```diff
@@ -442,9 +442,299 @@ public class MyApplication extends Application
...
+                in the MicroProfile Extension pack. See the Wildfly Admin Guide for more infomation:
```
Comment (asoldano) [inline docbook/reference/en/en-US/modules/Installation_Configuration.xml:592]: s/Wildfly/WildFly/

File: docbook/reference/en/en-US/modules/Installation_Configuration.xml [hunk 4]
```diff
@@ -442,9 +442,299 @@ public class MyApplication extends Application
...
+                <ulink url="https://docs.wildfly.org/19/Admin_Guide.html">https://docs.wildfly.org/19/Admin_Guide.html</ulink>.
+                Outside of Wildfly, an application will need the
```
Comment (asoldano) [inline docbook/reference/en/en-US/modules/Installation_Configuration.xml:594]: s/Wildfly/WildFly/

File: docbook/reference/en/en-US/modules/Installation_Configuration.xml (hunk 1 above)
Comment (ronsigal) [reply docbook/reference/en/en-US/modules/Installation_Configuration.xml:513]: @asoldano, wow. I don't think I ever noticed that!! ;-)

Comment (ronsigal) [reply docbook/reference/en/en-US/modules/Installation_Configuration.xml]: Done

File: docbook/reference/en/en-US/modules/Installation_Configuration.xml (hunk 2 above)
Comment (ronsigal) [reply docbook/reference/en/en-US/modules/Installation_Configuration.xml]: Done

File: docbook/reference/en/en-US/modules/Installation_Configuration.xml (hunk 4 above)
Comment (ronsigal) [reply docbook/reference/en/en-US/modules/Installation_Configuration.xml]: Done

**Pull request merged**